### PR TITLE
Feat/817 development of seedlot review step 1   applicant and seedlot information

### LINKED ISSUE #817: Development of Seedlot Review step 1 - applicant and seedlot information
Describe the task:

Create FE of Seedlot Review, Step 1 (applicant information) and integrate endpoints as you go along.

Figma design here: https://www.figma.com/file/ROgZJQezb1DDG4AvEgjtpw/Registration-Process----Reviewer?type=design&node-id=2744-284788&mode=design&t=XxxbormmPMPHmENb-4

Image

Acceptance Criteria
- [ ] Developed as per Figma
- [ ] relevant documentation updated
- [ ] tested and merged
- [ ] Develop floater save button place holder - design to be tested by TSC staff

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Closes #817 

Apologies for the huge PR.
A lot of refactoring and components were needed to achieve this.
I'm happy with the results, it's possible for that TSC super user 🦖 to edit Pending seedlots after all.

#### New
- Dev tools for dev/local env only 🙌, it's available on the profile side menu
    image
- Seedlot Review Access button, note that no RBAC is implemented as it'll make testing harder
    image

- Seedlot Review View page setup along with applicant and seedlot implemented
    `frontend/src/views/Seedlot/SeedlotReview/index.tsx`
    image

- `<Divider />` component for a horizontal line, it's using `<Row>` from carbon instead of `<hr>` from html, might have better with responsiveness
    image

- `<RowGap />` component for the grey row gap, using this is easier than setting the background to grey
    image

- `<EmailDisplay />` component for displaying and copying an email
    image
    image
-  `<DetailSection />` A section wrapper that provides a white container with rounded border and paddings. Intended for Seedlot Detail page, can be used elsewhere.

#### Changed
- Removed famLoginUser from local storage as it's not needed. To get JWT or other user info, use the dev tool.
- Refactored the `index.tsx` and `App.tsx`, the goal was to simplify the `index.tsx` file as conventionally it should be a specifying the root rendering element. 
- Refactored the logic for `SeedlotRegFormClassA`, separated the context out to `ContextContainerClassA` so it can be shared with Seedlot Review. Although some of the logic in `ContextContainerClassA` is not needed for the review page (yet), it is better to keep it there for now.
- Refactored the hierarchy for Edit Applicant Info page so it can be used in the review page as well.  
- Minor name changes

#### Removed
- None

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-22-frontend.apps.silver.devops.gov.bc.ca/)
[Backend](https://nr-spar-22-backend.apps.silver.devops.gov.bc.ca/actuator/health)
[Oracle-API](https://nr-spar-22-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

**Review by DerekRoberts — APPROVED** [review]
That's a hilarious amount of work for minimal visible change!  Glad it's done.  ;)

File: frontend/index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <link rel="icon" href="/favicon.ico" />
   <meta name="viewport" content="width=device-width, initial-scale=1" />
   <meta name="theme-color" content="#000000" />
-  <meta name="description" content="Web site created using create-react-app" />
+  <meta name="description" content="BC Seed Planning and Registry 2.0" />
```
Comment (RMCampos) [inline frontend/index.html:9]: 👍🏻

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -26,41 +31,73 @@ Amplify.configure(awsconfig);
...
-    </>
+    <ClassPrefix prefix={prefix}>
+      <ThemePreference>
+        <QueryClientProvider client={queryClient}>
+          <ToastContainer />
+          <RouterProvider router={signed ? signedRouter : notSignedRouter} />
+          <ReactQueryDevtools initialIsOpen={false} />
+        </QueryClientProvider>
+      </ThemePreference>
+    </ClassPrefix>
```
Comment (RMCampos) [inline frontend/src/App.tsx:100]: Refactored from index.tsx file to here. 👍🏻

File: frontend/src/components/Divider/styles.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@use '@bcgov-nr/nr-theme/design-tokens/variables.scss' as vars;
+
+.sub-section-divider {
+  height: 1px;
```
Comment (RMCampos) [inline frontend/src/components/Divider/styles.scss:4]: question (non-blocking): is px fine? Or should we change to rem?

File: frontend/src/components/MyProfile/index.tsx [hunk 4]
```diff
@@ -117,6 +118,28 @@ const MyProfile = () => {
           >
             Sign out
           </SideNavLink>
+          {
+            appVersion === 'dev'
+              ? (
+                <>
+                  <PanelSectionName title="Dev Zone" light />
```
Comment (RMCampos) [inline frontend/src/components/MyProfile/index.tsx:125]: Nice 👍🏻

**Review by RMCampos — APPROVED** [review]
Some optional things. That's a lot of work! Very nice job 🥇

File: frontend/src/components/Divider/styles.scss (hunk 3 above)
Comment (craigyu) [reply frontend/src/components/Divider/styles.scss:4]: oof, good catch

Comment (craigyu) [reply frontend/src/components/Divider/styles.scss]: https://github.com/bcgov/nr-spar/pull/922/commits/58d5338ee2824feca450f3e31b8ac68c5e407df4

File: frontend/src/components/LotApplicantAndInfoForm/index.tsx [hunk 5]
```diff
@@ -23,13 +23,15 @@ import {
 } from './constants';
 
 import './styles.scss';
+import Divider from '../Divider';
```
Comment (mgaseta) [inline frontend/src/components/LotApplicantAndInfoForm/index.tsx:26]: Add this import together with the other components imports.

File: frontend/src/components/ReadOnlyInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import './styles.scss';
+import { TextInput, TextInputSkeleton } from '@carbon/react';
```
Comment (mgaseta) [inline frontend/src/components/ReadOnlyInput/index.tsx:4]: Change this import to line 2 :)

File: frontend/src/config/FavouriteActivitiyMap.ts [hunk 7]
```diff
@@ -26,7 +26,7 @@ const FavouriteActivityMap: Record<string, FavActivityType> = {
     image: 'TableSplit',
     header: 'My Seedlots',
     description: 'Check and manage your own seedlots',
-    link: PathConstants.MY_SEEDLOTS,
+    link: ROUTES.MY_SEEDLOTS,
     highlighted: false
   },
   unkown: {
```
Comment (mgaseta) [inline frontend/src/config/FavouriteActivitiyMap.ts:32]: Non-related to the changes, but there is a typo here...

File: frontend/src/views/Seedlot/ContextContainerClassA/context.tsx
Comment (mgaseta) [inline frontend/src/views/Seedlot/ContextContainerClassA/context.tsx:1]: Don't even have a GIF for this, nice job! Context FTW!

File: frontend/src/views/Seedlot/ContextContainerClassA/index.tsx [hunk 8]
```diff
@@ -0,0 +1,676 @@
...
+  validateOrchardStep,
+  verifyOrchardStepCompleteness,
+  validateExtractionStep,
+  verifyExtractionStepCompleteness,
+  validateParentStep,
+  verifyParentStepCompleteness,
+  checkAllStepsCompletion,
+  getSeedlotPayload,
+  initEmptySteps,
+  resDataToState
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/ContextContainerClassA/index.tsx:57]: Just a suggestion: break this import in to less lines, maybe like 4 items per line, like on lines 11 - 14.

File: frontend/src/views/Seedlot/EditAClassApplication/index.tsx [hunk 9]
```diff
@@ -1,219 +1,33 @@
...
-
-import { getBreadcrumbs } from './utils';
 
 import './styles.scss';
-
-const EditAClassApplication = () => {
-  const navigate = useNavigate();
-  const { seedlotNumber } = useParams();
-
+import EditAClassApplicationForm from './Form';
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/EditAClassApplication/index.tsx:7]: Adjust import order.

File: frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx [hunk 10]
```diff
@@ -90,37 +91,36 @@ const FormProgress = (
...
+        <Column>
+          <Button
+            kind="tertiary"
+            size="md"
+            className="section-btn"
+            renderIcon={Edit}
+            onClick={() => navigate(addParamToPath(ROUTES.SEEDLOT_A_CLASS_REGISTRATION, seedlotNumber ?? ''))}
+            disabled={getSeedlotQueryStatus === 'loading'}
+          >
+            {seedlotStatusCode === 'SUB' ? 'Check your seedlot' : 'Edit seedlot form'}
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx:119]: We agreed to change "Check" to "View", right?

File: frontend/src/views/Seedlot/SeedlotReview/context.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/SeedlotReview/context.ts:1]: This can be safely removed, right?

**Review by mgaseta — CHANGES_REQUESTED** [review]
Phew, that was a lot... A LOT OF AMAZING CHANGES! Nice job @craigyu !
Added only a fez mark-ups, but nothing critical!

File: frontend/src/config/FavouriteActivitiyMap.ts (hunk 7 above)
Comment (craigyu) [reply frontend/src/config/FavouriteActivitiyMap.ts:32]: lol found more typos from me on this word, good eyes

File: frontend/src/views/Seedlot/ContextContainerClassA/index.tsx (hunk 8 above)
Comment (craigyu) [reply frontend/src/views/Seedlot/ContextContainerClassA/index.tsx:57]: ok, did 3 per line

**Comment by craigyu** [comment]
@mgaseta 
Tried to fix everything in this commit https://github.com/bcgov/nr-spar/pull/922/commits/55e79f7f34266a5750ffb648169066a26bb8a1b0

but then I forgot one thing so it's here https://github.com/bcgov/nr-spar/pull/922/commits/2809ba72f280baa8769db581707dac13710e34fc

**Review by mgaseta — APPROVED** [review]
LGTM! Great job! :1st_place_medal: :rocket:

**Pull request merged**